File: 0338-counting-bits/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* countBits(int n, int* returnSize){
    int * solution = calloc(n + 1, sizeof(int));
    *returnSize = n + 1;
    for (int i = 1; i < n + 1; i++) {
        solution[i] = solution[i >> 1] + (i & 1);
    }
    return solution;
}

File: 0338-counting-bits/solution.py
class Solution(object):
    def countBits(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        solution = [0] * (n + 1)
        for i in range(n+1):
            N = i
            while N != 0:
                solution[i] += N & 1
                N = N >> 1
        return solution
